ms2deepscore/models/EmbeddingEvaluatorModel.py

```diff
@@ -0,0 +1,285 @@
+                            val_losses.append(float(loss))
+                        print(f">>> Val_loss: {np.mean(val_losses):.6f}")
+
+                    evaluator_model.train()
+
+
+### Below is code for a InceptionTime sequence classification model
+        
+
+class InceptionTime(Module):
+    """
+    Default Inception Time model, typically used for time series or sequence classification.
+    """
+    def __init__(self, input_channels, output_channels, num_filters=32, **kwargs):
+        super().__init__()
+        self.inception_block = InceptionBlock(input_channels, num_filters, **kwargs)
+        self.global_avg_pool = nn.AdaptiveAvgPool1d(1)
+        self.fc = nn.Linear(num_filters * 4, output_channels)
+
+    def forward(self, x):
+        x = self.inception_block(x)
+        x = self.global_avg_pool(x).squeeze(-1)
+        x = self.fc(x)
+        return x
+
+
+class InceptionModule(Module):
+    def __init__(self, input_channels, num_filters, kernel_size=40, use_bottleneck=True):
+        super().__init__()
+        # Adjust kernel sizes to ensure they are odd for symmetric padding
+        kernel_sizes = [max(kernel_size // (2 ** i), 3) for i in range(3)]
+        kernel_sizes = [k - (k % 2 == 0) for k in kernel_sizes]
+
+        # Use bottleneck layer only if input_channels > 1
+        use_bottleneck = use_bottleneck if input_channels > 1 else False
+        self.bottleneck = nn.Conv1d(input_channels, num_filters, 1, bias=False) if use_bottleneck else nn.Identity()
+        
+        # Prepare convolutional layers with adjusted kernel sizes
+        conv_input_channels = num_filters if use_bottleneck else input_channels
+        self.convs = nn.ModuleList([nn.Conv1d(conv_input_channels, num_filters, k, 
+                                              padding="same") for k in kernel_sizes])
+        
+        # MaxPooling followed by a convolution
+        self.maxconvpool = nn.Sequential(
+            nn.MaxPool1d(3, stride=1, padding=1),
+            nn.Conv1d(input_channels, num_filters, 1, bias=False)
+        )
+        
+        # Batch normalization and activation function
+        self.bn = nn.BatchNorm1d(num_filters * 4)
+
+    def forward(self, x):
+        bottleneck_output = self.bottleneck(x)
+        conv_outputs = [conv(bottleneck_output) for conv in self.convs]
+        pooled_output = self.maxconvpool(x)
+        concatenated_output = torch.cat(conv_outputs + [pooled_output], dim=1)
+        return F.relu(self.bn(concatenated_output))
+
+
+class InceptionBlock(Module):
+    def __init__(self, input_channels, num_filters=32, use_residual=True, depth=6, **kwargs):
+        super().__init__()
+        self.use_residual = use_residual
+        self.depth = depth
+        self.inception_modules = nn.ModuleList()
+        self.shortcuts = nn.ModuleList()
+
+        for d in range(depth):
+            module_input_channels = input_channels if d == 0 else num_filters * 4
```

Comment:
It is a bit unclear what happens in these classes to me. For instance the num_filters *4. So I couldn't really check if everything makes sense here, but I also do not really know the InceptionTime model. Maybe a bit extra documentation could help.

Reply:
It is a quite complex architecture. I will add some more explanation to it, but mostly refer to the original paper.